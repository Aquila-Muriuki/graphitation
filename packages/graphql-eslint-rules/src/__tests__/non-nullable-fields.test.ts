import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../non-nullable-fields";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("non-nullable-fields", rule, {
  valid: [
    {
      code: `
type ChatMessage {
  id: String!
  content: String
}`,
      options: [{ fields: ["id"] }],
    },
  ],
  invalid: [
    {
      code: `
type ChatMessage {
  id: String
  content: String
  someOtherImportantField: String
}`,
      options: [{ fields: ["id", "someOtherImportantField"] }],
      errors: 2,
    },
  ],
});
