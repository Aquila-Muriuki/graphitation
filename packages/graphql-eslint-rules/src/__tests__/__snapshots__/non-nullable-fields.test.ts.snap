// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | type ChatMessage {
      2 |   id: String
      3 |   content: String
      4 |   someOtherImportantField: String
      5 | }

#### ⚙️ Options

    {
      "fields": [
        "id",
        "someOtherImportantField"
      ]
    }

#### ❌ Error 1/2

      1 | type ChatMessage {
    > 2 |   id: String
        |   ^^^^ Field "id" cannot be nullable.
      3 |   content: String

#### ❌ Error 2/2

      3 |   content: String
    > 4 |   someOtherImportantField: String
        |   ^^^^^^^^^^^^^^^^^^^^^^^^^ Field "someOtherImportantField" cannot be nullable.
      5 | }"
`;
